src/Countries/Australia.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Australia extends Country
+{
+    public function countryCode(): string
+    {
+        return 'au';
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'New Year\'s Day' => '01-01',
+            'Australia Day' => '01-26',
+            'Anzac Day' => '04-25',
+            'Christmas Day' => '12-25',
+            'Boxing Day' => '12-26',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))->setTimezone('Australia/Sydney');
+
+        return [
+            'Queen\'s Birthday' => CarbonImmutable::parse("second monday of june $year"),
```

Comment:
It's King's birthday now :)

It's also regional - different dates for different states.

Reply:
I've removed it. Once the package handles states we can get a little more granular.

Reply:
Yeah, there's something mentioning regional dates in the readme, but regional holidays in Australia get weird

Reply:
I've also removed Easter Saturday and Sunday as some states don't observe them.

Let's see what comes of @Nielsvanpach's PR #39.

Reply:
i've added documentation on how to handle regions: https://github.com/spatie/holidays?tab=readme-ov-file#contributing-a-new-country

Reply:
Why not just add all days, and suffix the description with the state in brackets? e.g. "King's Holiday (VIC)".

Reply:
I've added regional support to the state level.